iceoryx_hoofs/cmake/IceoryxPlatform.cmake

```diff
@@ -44,19 +44,28 @@ else()
 endif()
 
 if(CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
-    set(ICEORYX_WARNINGS PRIVATE ${ICEORYX_WARNINGS} /W0) # TODO iox-#33 set to /W1
+    set(ICEORYX_CXX_WARNINGS PRIVATE ${ICEORYX_CXX_WARNINGS} /W0) # TODO iox-#33 set to /W1
+    set(ICEORYX_CXX_WARNINGS PRIVATE ${ICEORYX_C_WARNINGS} /W0) # TODO iox-#33 set to /W1
     # todo: '/O2' and '/RTC1' (set by default) options are incompatible,
 elseif(CMAKE_CXX_COMPILER_ID MATCHES "GNU" OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-    set(ICEORYX_WARNINGS PRIVATE ${ICEORYX_WARNINGS} -W -Wall -Wextra -Wuninitialized -Wpedantic -Wstrict-aliasing -Wcast-align -Wno-noexcept-type -Wconversion)
+    set(ICEORYX_CXX_WARNINGS PRIVATE ${ICEORYX_CXX_WARNINGS} -W -Wall -Wextra -Wuninitialized -Wpedantic -Wstrict-aliasing -Wcast-align -Wno-noexcept-type -Wconversion)
+    set(ICEORYX_CXX_WARNINGS PRIVATE ${ICEORYX_C_WARNINGS} -W -Wall -Wextra -Wuninitialized -Wpedantic -Wstrict-aliasing -Wcast-align -Wconversion)
+
+    if(CMAKE_CXX_COMPILER_ID MATCHES "GNU")
+        set(ICEORYX_CXX_WARNINGS PRIVATE ${ICEORYX_CXX_WARNINGS} -Wuseless-cast)
+    endif()
 endif()
 
 if(BUILD_STRICT)
     if(CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
-        set(ICEORYX_WARNINGS ${ICEORYX_WARNINGS} /W0) # TODO iox-#33 set to /WX
+        set(ICEORYX_CXX_WARNINGS ${ICEORYX_CXX_WARNINGS} /W0) # TODO iox-#33 set to /WX
+        set(ICEORYX_CXX_WARNINGS ${ICEORYX_C_WARNINGS} /W0) # TODO iox-#33 set to /WX
     elseif(CMAKE_CXX_COMPILER_ID MATCHES "GNU")
-        set(ICEORYX_WARNINGS ${ICEORYX_WARNINGS} -Werror)
+        set(ICEORYX_CXX_WARNINGS ${ICEORYX_CXX_WARNINGS} -Werror)
+        set(ICEORYX_CXX_WARNINGS ${ICEORYX_C_WARNINGS} -Werror)
     elseif(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-        set(ICEORYX_WARNINGS ${ICEORYX_WARNINGS} -Werror)
+        set(ICEORYX_CXX_WARNINGS ${ICEORYX_CXX_WARNINGS} -Werror)
+        set(ICEORYX_CXX_WARNINGS ${ICEORYX_C_WARNINGS} -Werror)
```

Comment:
Also here
```cmake
        set(ICEORYX_C_WARNINGS ${ICEORYX_C_WARNINGS} -Werror)
        set(ICEORYX_CXX_WARNINGS ${ICEORYX_CXX_WARNINGS} -Werror)
```
Do you know why there is no `PRIVATE` in this statement?

Reply:
I would say `set` just defines a variable and therefore `PRIVATE` does not apply here. But when you take a look at `IceoryxPackageHelper.cmake` when those variables are applied then the `PRIVATE` keyword is added in `target_compile_options`

Reply:
No problem. I was just wondering why it was present in one place and not in another.